src/main/java/uk/gov/hmcts/reform/sendletter/entity/LetterRepository.java

```diff
@@ -31,6 +32,17 @@
 
     Optional<Letter> findById(UUID id);
 
+    @Query("select l.status from Letter l where l.id = :id")
+    Optional<LetterStatus> findStatusById(@Param("id") UUID id);
+
+    @Modifying(clearAutomatically = true)
```

Comment:
```
	/**
	 * Defines whether we should clear the underlying persistence context after executing the modifying query.
	 *
```

Reply:
without this, in tests, an old version of letter is fetched:
```
repository.saveAndFlush(letter);
// run the task
letter = repository.findById(letter.getId()).get(); // this letter was still in 'posted' status
```

Reply:
Can we not use @DirtiesContext on the test class ?